### PR TITLE
[M2M] Small fixes 2

### PR DESCRIPTION
**Pull request by ghost**
fixes
- cancel submit when enter key is pressed on filtering view
- cancel scrolling on up or down key

### CODE REVIEW

**Comment by ar4s** [comment]
:ok:

File: src/ralph/admin/autocomplete.py [hunk 1]
```diff
@@ -66,17 +67,7 @@ def get(self, request, *args, **kwargs):
...
-                'pk': obj.pk,
-                '__str__': str(obj),
-                'label': getattr(obj, 'autocomplete_str', None),
-                'edit_url': '{}?_popup=1'.format(
-                    get_admin_url(obj, 'change')
-                ) if can_edit else None,
-                'tooltip': getattr(obj, 'autocomplete_tooltip', None)
-            } for obj in self.get_queryset(request.user)
-        ]
+        results = self.get_results(request.user, can_edit)
```
Comment (mkurek) [inline src/ralph/admin/autocomplete.py:83]: shouldn't this method be implemented here as well (or `abc.abstractmethod`)?

File: src/ralph/admin/autocomplete.py [hunk 2]
```diff
@@ -189,6 +201,25 @@ def get_base_ids(self, model, value):
         queryset = self.get_query_filters(queryset, value, search_fields)
         return queryset[:self.limit].values_list('pk', flat=True)
 
+    def get_results(self, user, can_edit):
+        results = [
```
Comment (mkurek) [inline src/ralph/admin/autocomplete.py]: this list comprehension could be at least separated to another function (this is the same code as above).

File: src/ralph/admin/filters.py [hunk 3]
```diff
@@ -277,21 +279,28 @@ class RelatedAutocompleteFieldListFilter(RelatedFieldListFilter):
...
-                queryset = queryset.filter(**{self.field_path: value})
+        if not value:
+            ids = []
+        else:
+            ids = map(int, value.split(','))
+        q_param = models.Q()
+        for id_ in ids:
+            if id_ == self.empty_value:
+                q_param |= Q(**{'{}__isnull'.format(self.field_path): True})
+            q_param |= Q(**{self.field_path: id_})
```
Comment (mkurek) [inline src/ralph/admin/filters.py]: shouldn't this be in else block?

File: src/ralph/admin/autocomplete.py (hunk 1 above)
Comment (ghost) [inline src/ralph/admin/autocomplete.py:83]:
```
    def get_results(self, user, can_edit):                                                                                                                                                                                          
        raise NotImplementedError    
```

Would it be enough?

Comment (mkurek) [inline src/ralph/admin/autocomplete.py:83]: sure

File: src/ralph/admin/autocomplete.py [hunk 4]
```diff
@@ -92,25 +96,35 @@ def get_urls(self):
...
         class Detail(SuggestView):
+            empty_value = AUTOCOMPLETE_EMPTY_VALUE
             model = outer_model
 
             def dispatch(self, request, *args, **kwargs):
                 self.pk = request.GET.get(DETAIL_PARAM, None)
                 if not self.pk:
                     return HttpResponseBadRequest()
-                self.pks = self.pk.split(',')
+                self.values = list(map(int, self.pk.split(',')))
```
Comment (mkurek) [inline src/ralph/admin/autocomplete.py]: i'm not sure if mapping to int is good idea here - what if we'll have uid somewhere in the future?

Comment (ghost) [inline src/ralph/admin/autocomplete.py]: Then much more should be changed, I'd rather not do it now.

**Comment by mkurek** [comment]
:ok:

**Comment by szok** [comment]
:ok:

**Pull request merged**